File: spring-boot-entity-example/src/main/java/io/beaniejoy/entityexample/entity/cafe/CafeMenu.java
package io.beaniejoy.entityexample.entity.cafe;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "cafe_menu")
public class CafeMenu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private BigDecimal price;

    @ManyToOne
    @JoinColumn(name = "cafe_id")
    private Cafe cafe;

    @OneToMany(mappedBy = "cafeMenu", fetch = FetchType.LAZY)
    private List<MenuOption> menuOptions;
}

File: spring-boot-v3-batch-demo/src/main/kotlin/io/beaniejoy/batch/domain/entity/CafeMenu.kt
package io.beaniejoy.batch.domain.entity

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "cafe_menus")
class CafeMenu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null

    private val name: String? = null

    private val price: BigDecimal? = null

    @ManyToOne
    @JoinColumn(name = "cafe_id")
    private val cafe: Cafe? = null
}